server/src/main/java/org/opensearch/index/query/IntervalsSourceProvider.java

```diff
@@ -630,6 +634,158 @@ String getUseField() {
         }
     }
 
+    public static class Regexp extends IntervalsSourceProvider {
+
+        public static final String NAME = "regexp";
+        public static final int DEFAULT_FLAGS_VALUE = RegexpFlag.ALL.value();
+
+        private final String pattern;
+        private final int flags;
+        private final String useField;
+        private final Integer maxExpansions;
+
+        public Regexp(String pattern, int flags, String useField, Integer maxExpansions) {
+            this.pattern = pattern;
+            this.flags = flags;
+            this.useField = useField;
+            this.maxExpansions = (maxExpansions != null && maxExpansions > 0) ? maxExpansions : null;
+        }
+
+        public Regexp(StreamInput in) throws IOException {
+            this.pattern = in.readString();
+            this.flags = in.readVInt();
+            this.useField = in.readOptionalString();
+            this.maxExpansions = in.readOptionalVInt();
+        }
+
+        @Override
+        public IntervalsSource getSource(QueryShardContext context, MappedFieldType fieldType) {
+            IntervalsSource source;
+
+            final RegExp regexp = new RegExp(pattern, flags);
```

Comment:
I just spent several minutes trying to find the 2-argument constructor for the nested "Regexp" class before realizing this (RegExp) is a different class all together. I don't have a good suggestion beyond using a fully qualified import here are renaming Regexp and breaking the symmetry between the nested class name and the field being parsed. Feel free to keep as-is unless you think an alternative is better.

Reply:
I will use fully qualified name.